internal/services/cdn/cdn_frontdoor_route_resource.go

```diff
@@ -0,0 +1,533 @@
+package cdn
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/parse"
+	track1 "github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/sdk/2021-06-01"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/cdn/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceCdnFrontdoorRoute() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceCdnFrontdoorRouteCreate,
+		Read:   resourceCdnFrontdoorRouteRead,
+		Update: resourceCdnFrontdoorRouteUpdate,
+		Delete: resourceCdnFrontdoorRouteDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.FrontdoorRouteID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"cdn_frontdoor_endpoint_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FrontdoorEndpointID,
+			},
+
+			"cdn_frontdoor_origin_group_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.FrontdoorOriginGroupID,
+			},
+
+			"cdn_frontdoor_origin_ids": {
+				Type:     pluginsdk.TypeList,
+				Required: true,
+				ForceNew: true,
+
+				Elem: &pluginsdk.Schema{
+					Type:         pluginsdk.TypeString,
+					ValidateFunc: validate.FrontdoorOriginID,
+				},
+			},
```

Comment:
What's the purpose of `cdn_frontdoor_origin_ids`? Isn't this conflicting with `cdn_frontdoor_origin_group_id` which contains origins as well?

Reply:
@heoelri the `cdn_frontdoor_origin_ids` are only exposed for the create workflow logic. If I didn't make the route reference the origin ids Terraform would try to create both of the resources at the same time and the route resource would fail because the origin didn't exist yet.